src/Giraffe/EndpointRouting.fs

```diff
@@ -13,7 +13,7 @@ open Giraffe
 
 module private RouteTemplateBuilder =
     let private guidPattern =
-        "([0-9A-Fa-f]{{8}}\-[0-9A-Fa-f]{{4}}\-[0-9A-Fa-f]{{4}}\-[0-9A-Fa-f]{{4}}\-[0-9A-Fa-f]{{12}}|[0-9A-Fa-f]{{32}}|[-_0-9A-Za-z]{{22}})"
+        "^[0-9A-Fa-f]{{8}}-[0-9A-Fa-f]{{4}}-[0-9A-Fa-f]{{4}}-[0-9A-Fa-f]{{4}}-[0-9A-Fa-f]{{12}}$|^[0-9A-Fa-f]{{32}}$|^[-_0-9A-Za-z]{{22}}$"
```

Comment:
Maybe add a comment on why this is done with Regex in the first place.
Without much context, I would assume Guid.TryParse would do the trick here.

Reply:
Indeed, my initial approach was to try to use the `Guid.TryParse`, but was not able. If this endpoint worked with normal Guid, it would only require a [route constraint](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/routing?view=aspnetcore-8.0#route-constraints) like `{id:guid}`, but in order to be compatible with Giraffe custom router we must support [ShortGuid](https://github.com/giraffe-fsharp/Giraffe/blob/master/src/Giraffe/ShortGuid.fs) too.

Reply:
Added the comment with this commit https://github.com/giraffe-fsharp/Giraffe/pull/594/commits/d7188c7aa44e6688516fa74566876a2646cd3b4b

Reply:
This same regex (+/-) is used here https://github.com/giraffe-fsharp/Giraffe/blob/master/src/Giraffe/FormatExpressions.fs#L30-L31